addons/qodot/game_definitions/fgd/point_classes/light.gd

```diff
@@ -2,46 +2,49 @@
 class_name QodotLight
 extends QodotEntity
 
+const float_tolerance = 0.0001
 var light_node: Light3D = null
 
 func update_properties():
 	if not Engine.is_editor_hint():
 		return
-	
+
 	for child in get_children():
 		remove_child(child)
 		child.queue_free()
-	
-	if 'mangle' in properties and (properties['mangle'] as Vector3).x != 0.005:
+
+	var mangle_x = (properties['mangle'] as Vector3).x
```

Comment:
This needs to be under `if 'mangle' in properties`, otherwise it runs an error if `properties['mangle']` doesn't exist (which is possible, however unlikely).

Additionally, this would be better off grabbing the whole Vector3 value, as you need to access the rest of `properties['mangle']` anyway.

Also, if `float_tolerance` is gonna be a constant but we're already comparing mangle.x against a value, why not just modify the 0.005 to 0.0049 and get rid of `float_tolerance` or change `float_tolerance` to 0.0049 and get rid of the hardcoded 0.005 comparison?

Reply:
> This needs to be under `if 'mangle' in properties`, otherwise it runs an error if `properties['mangle']` doesn't exist (which is possible, however unlikely).
>
> Additionally, this would be better off grabbing the whole Vector3 value, as you need to access the rest of `properties['mangle']` anyway.

Oh, you're right.
I can fix it, but in that case I suppose I could do:

```gdscript
if 'mangle' in properties and !((properties['mangle'] as Vector3).x > 0.005 - float_tolerance and (properties['mangle'] as Vector3).x < 0.005 + float_tolerance):
  bla bla bla 
```

It looks horrible, but that way I wouldn't add an extra nesting level to the `if`.

Moving the `mangle` variable inside the `if` is all the same to me, but I don't know which would comply better with the coding style for this project.


> Also, if `float_tolerance` is gonna be a constant but we're already comparing mangle.x against a value, why not just modify the 0.005 to 0.0049 and get rid of `float_tolerance` or change `float_tolerance` to 0.0049 and get rid of the hardcoded 0.005 comparison?

I thought about that, but I thought adding the tolerance would make more clear I was comparing to the `0.005` set in `light_base_class.tres`.

Same as before: I have no problem changing that to whatever suits the project better.